core/cap-0063.md

```diff
@@ -0,0 +1,436 @@
++    uint32 diskReadBytes;
+     // The maximum number of bytes this transaction can write to ledger
+     uint32 writeBytes;
+ };
+ 
++struct SorobanResourcesExtV0
++{
++    // Vector of uint16_t indices representing what Soroban
++    // entries in the footprint are evicted, based on the
++    // order of keys provided in the readWrite footprint.
++    opaque<> evictedSorobanEntries;
++};
++
+ // The transaction extension for Soroban.
+ struct SorobanTransactionData
+ {
+-    ExtensionPoint ext;
++    union switch (int v)
++    {
++    case 0:
++        void;
++    case 1:
++        SorobanResourcesExtV0 resourceExt;
++    } ext;
+     SorobanResources resources;
+     // Amount of the transaction `fee` allocated to the Soroban resource fees.
+     // The fraction of `resourceFee` corresponding to `resources` specified 
+```
+
+### Semantics
+
+#### In-memory vs Disk Read Resources
+
+Read resources are now separated between memory backed and disk backed state as follows:
+
+- in-memory entries:
+  - Live Soroban entries
+  - Archived entries that have not yet been evicted (i.e. still exist in the live BucketList)
+- disk entries:
+  - Archived Soroban entries that have been evicted
+  - Classic entries
+
+In-memory entries are subject to the corresponding in-memory limits, but charge no entry or byte based
+read fees. On disk entries are subject to the corresponding limits and also charge entry and byte based
+read fees.
+
+Note that while read resources and fees have changed, write semantics have not. Writes to both entry
+types have the same fee and limits as protocol 22.
+
+#### Initial Network Config Settings
+
+Initially, all disk read limits and fees should match exactly the limits and fees of reads today. Limits must not
+decrease from current values in order to prevent bricking existing contracts. In the worst case, a contract can read
+up to `txMaxReadEntries - 2` classic entries today (all classic entries + wasm + instance). Current limits have been
+carefully measured assuming disk access only, so the addition of in-memory state should not negatively affect the network
+in any way should we maintain these values via disk read limits.
+
+TODO:  In-memory read entries must be measured in order to find a reasonable starting point.
+
+#### `evictedSorobanEntries` Vector
+
+For the purposes of block creation, the `LedgerKey` alone is sufficient to distinguish between classic and Soroban data.
+However, a disk read would be required prior to tx application in order to determine if a Soroban entry was live or evicted.
+This is necessary for determining what resources limits a Soroban key counts towards, but
+exposes a significant DOS angle. To prevent this attack, the footprint must statically declare which entries are live vs. evicted.
+
+This is accomplished via the `evictedSorobanEntries` vector. If any evicted Soroban entry is present in a TX's footprint,
```

Comment:
what happens if you mark a ledger entry as evicted in the invoke host function operation but during application it turns out the ledger entry has already been restored? will the operation still succeed?

I assume that if a ledger entry is not present in  `evictedSorobanEntries` and it turns out that the ledger entry is evicted, the transaction will fail regardless of whether there are enough fees to pay for the unexpected write. Is that correct?

Reply:
Yeah, if you don't mark an entry as evicted you fail. If you do mark it as evicted and it's not, you're spending more in fees and resources, so we shouldn't fail. I'll update this section, but if an entry is marked evicted, we should charge it as if it's evicted no matter what, but not fail if it's actually live.

Reply:
See added "Incorrect Entries in `evictedSorobanEntries` Vector" section.